src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp

```diff
@@ -152,6 +153,51 @@ class instance_zulaman : public InstanceMapScript
                 }
             }
 
+            void OnUnitDeath(Unit* unit) override
+            {
+                InstanceScript::OnUnitDeath(unit);
+
+                Creature* creature = unit->ToCreature();
+                if (!creature)
+                    return;
+
+                if (creature->HasStringId("NalorakkWave1"))
```

Comment:
the stringid is overkill if you ask me, you can "simply" check if the creature is within the related group

```
if (SpawnMetadata const* data = sObjectMgr->GetSpawnMetadata(SpawnObjectType::SPAWN_TYPE_CREATURE, me->GetSpawnId()))
            if (data->spawnGroupData && data->spawnGroupData->groupId == SPAWN_GROUP_MOJO)
```

maybe there is an easier way to access groupId on this context though (InstanceMapScript)